frontend/src/app/main/data/helpers.cljs

```diff
@@ -168,3 +169,24 @@
   [state]
   (when-let [{:keys [x y width height]} (get-in state [:workspace-local :vbox])]
     (gpt/point (+ x (/ width 2)) (+ y (/ height 2)))))
+
+(defn- name-seq
+  "Creates a lazy sequence of names in the pattern:
+   (str name),
+   (str name (suffix-fn 1)),
+   (str name (suffix-fn 2)),
+   (str name (suffix-fn 3))..."
+  [base-name suffix-fn]
+  (concat
+   [base-name]
+   (map #(str/concat base-name (suffix-fn %))
+        (iterate inc 1))))
+
+(defn get-unique-copy-name
+  "Returns first unique name from the sequence that doesn't exist in existing-names"
+  [source-name existing-names suffix-fn]
+  (let [existing-name-set (set existing-names)
+        names (name-seq source-name suffix-fn)]
+    (->> names
+         (remove #(contains? existing-name-set %))
+         first)))
```

Comment:
We already have a method for doing that, maybe it is not as complete as this one but I think we can improve the current approach instead of creating a new one just for this https://github.com/penpot/penpot/blob/develop/common/src/app/common/files/helpers.cljc#L408